nodejs/nodejs-cloud-run-hello-world/test/system.test.js

```diff
@@ -0,0 +1,20 @@
+const assert = require('assert');
+const {request} = require('gaxios');
+
+const port = process.env.PORT || '8080';
+const url = process.env.SERVICE_URL || `http://localhost:${port}`;
+
+describe('Hello World', () => {
+  it('can respond to an HTTP request', async () => {
+    console.log(`    - Requesting GET ${url}/...`)
```

Comment:
what I used to do, which I thought worked really well, would be to actually stand up and tear down the web application in the `before` statement of the test; I believe you might be able to bind to `0` (maybe?) and get a random port, and there should be a method `express` exposes for terminating the server.

Reply:
In this case, I'm testing against the service running on Cloud Run, and for reasons of 1. Developer Education, 2. Lack of a graceful way to handle Cloud Run deploys the setup/teardown is externalized. I have ideas around a hand-written library in the future happy to discuss